xwiki-platform-core/xwiki-platform-eventstream/xwiki-platform-eventstream-api/src/main/java/org/xwiki/eventstream/events/AbstractEventStreamEvent.java

```diff
@@ -27,6 +27,12 @@
  */
 public abstract class AbstractEventStreamEvent implements org.xwiki.observation.event.Event
 {
+    /**
+     * Used to provide a key to a property in the current execution context that avoids stepping into a loop when
+     * triggering new events.
+     */
+    public static final String EVENT_LOOP_CONTEXT_LOCK_PROPERTY = "eventLoopContextLockProperty";
```

Comment:
bq.  Also, it would mean that we’re using internal properties of the Event Stream from the Activity Stream module.

This is not really an issue IMO. You are talking about two things located in the same module right now. But nobody else is supposed to used that property since you did what was required to protect against cycles already. Making an internal tool public is easy but the opposite is very hard so without any use case in mind you should start internal.